### ISSUE TITLE
测试速度有点慢

### ISSUE THREAD

**Issue by HayashiKatsura** [hyao1/GLAD#23]
作者您好，我在测试的过程中，发现速度有点慢，请问您知道可能时什么原因吗？代码走的是main.py下的visual里，具体慢的代码是在下面这部分
 “”“  
        scores = normalize(np.array(scores))
        labels = np.array(labels)
        preds = np.array(preds)
        masks = np.array(masks, dtype=np.int_)

        precisions_image, recalls_image, _ = precision_recall_curve(labels, scores)
        f1_scores_image = (2 * precisions_image * recalls_image) / (precisions_image + recalls_image)
        best_f1_scores_image = np.max(f1_scores_image[np.isfinite(f1_scores_image)])
        auroc_image = roc_auc_score(labels, scores)
        AP_image = average_precision_score(labels, scores)

        precisions_pixel, recalls_pixel, _ = precision_recall_curve(masks.ravel(), preds.ravel())
        f1_scores_pixel = (2 * precisions_pixel * recalls_pixel) / (precisions_pixel + recalls_pixel)
        best_f1_scores_pixel = np.max(f1_scores_pixel[np.isfinite(f1_scores_pixel)])
        auroc_pixel = roc_auc_score(masks.ravel(), preds.ravel())
        AP_pixel = average_precision_score(masks.ravel(), preds.ravel())

        pro = compute_pro(masks, preds)

 “”“  
数据集为MvtecAD，类别为all，15个类别，参数如下，和提供的完全一致，
"--checkpointing_steps", "500",
 "--instance_prompt","a photo of sks",
 "--resolution", "512",
"--test_batch_size","2",
 "--class_name", "all",
 "--pre_compute_text_embeddings",
 "--seed", "0"

GPU为单张3090 24g
快的话这部分代码有的类别需要21分钟，慢的话要有的类别接近1个半小时。全部测完，快一天了要

**Comment by hyao1** [comment]
21分钟左右是正常的，由于我们的方法在生成中需要在每一步都需要进行比较，测试的时候确实会比较慢，可以考虑较少推理步数加速。

**Comment by hyao1** [comment]
在我自己在2080ti上测试的时候，确实需要20分钟左右。但是我没有具体测试过你列出代码的用时，这一部分代码是我用的numpy和sklearn.metrics库中的函数，以及DDAD中计算compute_pro的代码。可能是因为这一部分代码主要是在cpu上运行的，所以会比较慢。

**Comment by hyao1** [comment]
我自己在2080ti上测试完所有类别最多也就一个小时左右。这一部分代码主要是在cpu上运行的，你可以看看是不是自己的cpu占用比较多。

**Comment by HayashiKatsura** [comment]
> 我自己在2080ti上测试完所有类别最多也就一个小时左右。这一部分代码主要是在cpu上运行的，你可以看看是不是自己的cpu占用比较多。

明白了，感谢
